App/Program.cs

```diff
@@ -26,7 +27,23 @@ private static void Main(string[] args)
         try
         {
             ComWrappersSupport.InitializeComWrappers();
-            if (!CheckSingleInstance()) return;
+            AppInstance mainInstance = GetMainInstance();
+            if (!mainInstance.IsCurrent)
+            {
+                var activationArgs = AppInstance.GetCurrent().GetActivatedEventArgs();
+                mainInstance.RedirectActivationToAsync(activationArgs).AsTask().Wait();
+                return;
+            }
+
+            // Register for URI handling (known as "protocol activation")
+#if DEBUG
+            const string scheme = "coder-debug";
+#else
+            const string scheme = "coder";
+#endif
+            var thisBin = Assembly.GetExecutingAssembly().Location;
+            ActivationRegistrationManager.RegisterForProtocolActivation(scheme, thisBin + ",1", "Coder Desktop", "");
```

Comment:
Is it typical for apps to register themselves? Or is it more common for it to be registered in the installer (not sure if this is even an option)?

Reply:
The installer could do it, but I'm not sure it provides any advantage.

One advantage of the App registering itself is that you don't need to install to debug, which is nice.

We might want to cosider _unregistering_ in the uninstaller, which I can look into in another PR.